include/sys/arc_impl.h

```diff
@@ -240,6 +240,9 @@ struct arc_buf_hdr {
 	uint64_t		b_birth;
 
 	arc_buf_contents_t	b_type;
+	uint8_t			b_complevel;
+	uint8_t			b_reserved1; /* reserved for future use */
+	uint16_t		b_reserved2; /* reserved for future use */
```

Comment:
Reserved for what?

Reply:
taken from allans reference implementation

Reply:
i'm not sure if i can remove it. is it relevant to the physical filesystem structure? if yes, it may destroy backward compatiblity to previous versions of the zstd implementation which could kill my installations. i know this sounds egoistic, but other people might use it already too

Reply:
Hmm, well, normally we should not care for people using unstable/no-yet-merged stuff on production systems ;) I‘d say we can test it... I will do a test today

Reply:
It's for future use, not previous use...

Reply:
if its may create fs layout conflicts, its sane to keep reserved bytes for future use to avoid new conflicts for possible future changes. if it does not have any influence to the layout, we can waste it, since its irrelevant

Reply:
but there is another argument. there is a reason why he added 3 bytes reserved after the byte complevel. does someone know the magic word "alignment" ? that really important. especially on mips cpu's the compiler has to realigned the code otherwise and this makes the code slow. and unaligned accessess are slow on x86 too, but on x86 the cpu can handle unaligned addresses. its just slower

Reply:
in this case just add a comment or name it `alingment`

Reply:
true

Reply:
another solution is using a int32 field.

Reply:
Maybe @allanjude could comment on this, too? Before keeping something for performance reasons, this should be tested with and without IMHO.

Reply:
the performance impact will not be measureable on common architectures. only on mips we will see a impact for all accesses on fields comming after. its also uncommon to insert such fields at the beginning and not at the end. but i will only change this if it has no fs layout impact. and using a int32 field is the more clean solution anyway

Reply:
@mschilli87
It's not a problem keeping it at all, if it might be a future problem adding it back in, I say keep it.
Just like... Allan did in the first place.

@brainslayer

> but i will only change this if it has no fs layout impact
It has no fs layout impact on master either way. We (as a community) are not supposed to have change in a PR for our own convenience...